### ISSUE TITLE
Rosetta 2 translation layer for Asahi?

### ISSUE THREAD

**Issue by ghostforest** [AsahiLinux/docs#131]
Hi I'm wondering about a few things.

If I see this correctly Asahi Linux is based on arch but offers Fedora like packages?

Will there be a rosetta2 like translation layer for running non arm software on arm?

Is there a more suited chat to ask stuff like this?

Thank you

**Comment by Rastafabisch** [comment]
early on in the development process, Asahi used to be Arch based. While you can still install the Arch version, it's not recommend anymore as it's not updated with new features as regularly as the Fedora based releases. 
There are several tools to run x86/64 binaries on Linux. However they are not exactly a drop in replacement to Rosetta 2. For one, they do not translate binaries like rosetta but rather provide a runtime environment, more alike to emulation. While Rosetta 2 can emulate code as well, if necessary, port of its speed is the translation on installation/first run. Furthermore a rather tedious setup process is required on Linux, since Linux can't dynamically switch between the ways x86/64 talk to memory compared to how apple silicon and some other modern arm CPUS like the Raspberry Pi 5 addresses memory.

However many of the most "mainstream" applications can already be run nativly without any emulation requirement, especially  if their source code is available. Some still require some work, though. 

If you want to run windows software on Asahi, I would suggest staying on macOS for now, where Wine "just works", while on Asahi the setup involves setting up special purpose microVMs. It's not exactly consumer ready just yet.

**Comment by mkurz** [comment]
> While you can still install the Arch version, it's not recommend anymore as it's not updated with new features as regularly as the Fedora based releases.

I have to disagree partly. I run Arch Linux ARM since more than two years and my system is totally up to date, having the same features like Asahi Fedora (speakers, latest opengl, etc.). This is possible thanks to [joske's pull requests](https://github.com/AsahiLinux/PKGBUILDs/pulls/joske). So it is possible to keep using Arch.

**Comment by ghostforest** [comment]
Thanks for clarification. I think it would be cool to have Fedora and Arch as possible Linux options.

Another question: how good are the apple silicon GPU drivers? Crossover or Wine on Mac I feel is still inferior to Wine on Linux since linux supports Vulkan. Would I have benefits there from using Ashai on an apple silicon machine or are the drivers not as developed rn?

**Comment by a-plastic-bag** [comment]
> Thanks for clarification. I think it would be cool to have Fedora and Arch as possible Linux options.
> 
> Another question: how good are the apple silicon GPU drivers? Crossover or Wine on Mac I feel is still inferior to Wine on Linux since linux supports Vulkan. Would I have benefits there from using Ashai on an apple silicon machine or are the drivers not as developed rn?

There's no publicly available vulkan driver for asahi rn, they're working on it and it's pretty close afaik but it needs some extra features in order to work with Wine. Once that drops, Wine gaming should improve greatly. In the meantime though, it's probably more streamlined to just stick to macOS until there is an official method available for Wine and the likes.

**Comment by gallantcrusader** [comment]
as an update, muvm is doing pretty good

**Comment by davide125** [comment]
See https://docs.fedoraproject.org/en-US/fedora-asahi-remix/x86-support/
